tools/wptrunner/wptrunner/browsers/base.py

```diff
@@ -275,3 +280,117 @@ def __call__(self, line):
         self.logger.process_output(self.pid,
                                    line.decode("utf8", "replace"),
                                    command=" ".join(self.command) if self.command else "")
+
+
+class WebDriverBrowser(Browser):
+    __metaclass__ = ABCMeta
+
+    default_base_path = "/"
+
+    def __init__(self, logger, binary, webdriver_binary, webdriver_args=None,
+                 host="127.0.0.1", port=None, base_path="", env=None, **kwargs):
+        super().__init__(logger)
+
+        if webdriver_binary is None:
+            raise ValueError("WebDriver server binary must be given "
+                             "to --webdriver-binary argument")
+
+        self.logger = logger
+        self.binary = binary
+        self.webdriver_binary = webdriver_binary
+
+        self.host = host
+        self._port = port
```

Comment:
`self.port` will be read later in this function, so rather than the `port` property doing it, can we just use `get_free_port()` here? Less indirection, easier to understand.

Reply:
I see. Well let's leave it alone.